### ISSUE TITLE
Wrong minVersion & maxVersion for Language Packs

### ISSUE THREAD

**Issue by JustOff** [Pale-Moon-Addons-Team/phoebus#57]
While testing 27.4.0RC1 I found that Phoebus [always returns](https://github.com/Pale-Moon-Addons-Team/phoebus/blob/RELEASE/phoebus/components/aus/aus.php#L105) `{27.0.0a1:27.*}` as `{minVersion:maxVersion}` for Language Packs, that allows to install incompatible LPs regardless of the [measures we have taken to prevent it](https://github.com/MoonchildProductions/Pale-Moon/pull/1068#issuecomment-298229829).

We should fix this before 27.4 release.

**Comment by mattatobin** [comment]
While this is theoretically possible.. XPInstall should still reject or approve based on install.rdf but AUS should be updated to match the new policy..

**Comment by JustOff** [comment]
Unfortunately it does not rejects, this is how I ran into it.

**Comment by mattatobin** [comment]
Well I dunno why strict doesn't mean strict anymore.. Do you have compatibility checks disabled or something?

Regardless, file a PR against the RELEASE branch to the proper values in AUS.

**Comment by JustOff** [comment]
PM 27.4.0 RC1, clean profile, it allows to install any of LPs 27.3.1 right now.

**Comment by mattatobin** [comment]
Do you think we need to specify each min and max version in the array? Seems to be all the same..

**Comment by JustOff** [comment]
Honestly I'm not sure so just decided to provide maximum flexibility. You think this is overkill?

**Comment by JustOff** [comment]
Well, let's use common version limits for all LPs. We can always split them if we need.

**Comment by mattatobin** [comment]
You can't have the version numbers in the array like that.. Now, minAppVersion and maxAppVersion are valid add-on IDs to AUS and the website..

It will have to be specified either for each language pack like you had it OR done the way it originally was.

**Comment by JustOff** [comment]
Oops... I just tried to put this settings in one module to avoid future mistakes on update. Can we simply define new array `$arrayLangPackAppLimits` for that?

**Comment by mattatobin** [comment]
Why not put it back how it was with the correct values...

**Comment by JustOff** [comment]
I'm just trying to keep all variables in one place to simplify update procedure.

**Comment by mattatobin** [comment]
Whatever man.. Here is what you are gonna do.. I want a clean patch so you are gonna revert your branch to RELEASE state and commit your FINAL changes. DO not create a third pull request..

IN THE FUTURE.. I want to know and approve changes before pull requests are made. I would like to prevent garbage history as much as possible.

**Comment by mattatobin** [comment]
Here is the deal.. You can have the values in AUS or you can have them in each langpack data structure..

**Comment by JustOff** [comment]
I think we should ask @wolfbeast which option is preferred, since he is the one who carries out LPs updates.

**Comment by mattatobin** [comment]
In the meantime.. See if you can find out why strict isn't strict anymore for extensions updating via AUS over in the Pale Moon codebase because this behavior regardless of what AUS says should still be obey'd as strict maxVersion on that side.

**Comment by wolfbeast** [comment]
As far as I'm concerned, I really don't care what is done on the server side, as long as the version range offered is in line with the visiting client. Figure something out that works. Try not to bicker too much.

More importantly, yes, strict compat should be honored in the browser. I don't understand why it isn't.

**Comment by JustOff** [comment]
> I really don't care what is done on the server side, as long as the version range offered is in line with the visiting client

@wolfbeast, you will have to manually change maxVersion for each new release, so the question is to do it in a [separate file](https://github.com/Pale-Moon-Addons-Team/phoebus/blob/RELEASE/phoebus/components/aus/aus.php#L106) or in the same one, where the language packs are described. And in case of last choice - to do it in a [single line](https://github.com/Pale-Moon-Addons-Team/phoebus/pull/59/files#diff-5d31d909a7e347151aacf646ee700253) or in [each langpack data structure](https://github.com/Pale-Moon-Addons-Team/phoebus/pull/59/commits/ddeeb8adba01e581ec7df188b29011c1b8949b54#diff-5d31d909a7e347151aacf646ee700253). I have no reason to bicker, I'm ready to align the PR as you and Tobin decide.

**Comment by JustOff** [comment]
> See if you can find out why strict isn't strict anymore

> strict compat should be honored in the browser. I don't understand why it isn't.

As far as I remember such behaviour was introduced in times of Firefox 4 or 5  (before extensions were made non-strict by default) to allow to override maxVersion on the server side without re-upload the new xpi. I'll look more closely how easy this can be changed without breaking anything.

**Comment by wolfbeast** [comment]
@JustOff looking at the options and wanting to minimize the release overhead, it should most definitely be a single location. I don't want to have to do it for each individual language pack.
In addition, having a clear array instead of an obscure code location to edit something is better. inside the code, referring to such things as variables is always better than hard-coding. That's all I have to say about it, you two figure out the details.

**Comment by JustOff** [comment]
> WHAT are the minimum and maximum versions for the language packs.. ARE they truly backwards compatible all the way back to 27.0.0?

Yes.

> looking at the options and wanting to minimize the release overhead, it should most definitely be a single location. I don't want to have to do it for each individual language pack.

@mattatobin this is exactly what is currently proposed in #59, are you ok with this? 

>  I want a clean patch so you are gonna revert your branch to RELEASE state and commit your FINAL changes.

And since you have edited your previous message, I want to clarify, is it still necessary to combine commits into one?

**Comment by mattatobin** [comment]
I'll just take it as is.. I'll approve and send to LIVE when I get home.. Or Ryan can.

**Comment by mattatobin** [comment]
Alright @JustOff it is sitting on dev.addons.palemoon.org please test aus by prefixing the domain for the two AUS prefs with dev dot.

@Lootyhoof This is a merge blocker - Please do not merge RELEASE into LIVE until this is resolved.

**Comment by JustOff** [comment]
I confirm that everything works correctly on `dev.addons.palemoon.org`:
* The [update link](https://dev.addons.palemoon.org/?component=aus&reqVersion=2&id=langpack-ru@palemoon.org&version=27.3.1&maxAppVersion=27.3.*&status=userDisabled,incompatible&appID={8de7fcbb-c55c-4fbe-bfc5-fc555c87dbc4}&appVersion=27.4.0&appOS=WINNT&appABI=x86-msvc&locale=en-US&currentAppVersion=27.4.0&updateType=49&compatMode=normal&reqVersion=2&id=langpack-ru@palemoon.org&version=27.3.1&maxAppVersion=27.3.*&status=userDisabled,incompatible&appID={8de7fcbb-c55c-4fbe-bfc5-fc555c87dbc4}&appVersion=27.4.0&appOS=WINNT&appABI=x86-msvc&locale=en-US&currentAppVersion=27.4.0&updateType=49&compatMode=normal) returns the expected data (27.0.0-27.3.0).
* The change of `extensions.update.background.url` and `extensions.update.url` to `dev.addons.palemoon.org` leads that incompatible LP can't be no more instaled.

**Comment by mattatobin** [comment]
Alright.. merged into LIVE it will be picked up at the top of the hour.

AS AN ADDITIONAL.. Add the following to the install.rdf for all the langpacks:
`<em:strictCompatibility>true</em:strictCompatibility>`

Why? Because there's no kill like overkill.

**Comment by JustOff** [comment]
Thanks for cooperation.
